src/Fetch/Message.php

```diff
@@ -670,7 +670,7 @@ protected function processAddressObject($addresses)
         $outputAddresses = array();
         if (is_array($addresses))
             foreach ($addresses as $address) {
-                if (property_exists($address, 'mailbox') && $address->mailbox != 'undisclosed-recipients') {
+                if (property_exists($address, 'mailbox') && !empty($address->host)) {
```

Comment:
Daniels check only makes sense if changes the order:
```php 
if (property_exists($address, 'mailbox') && property_exists($address, 'host') && !empty($address->host)) 
{

}
```
´Then it would be fine...